api/core/tools/provider/builtin/tavily/tools/tavily_search.py

```diff
@@ -24,87 +24,34 @@ class TavilySearch:
     def __init__(self, api_key: str) -> None:
         self.api_key = api_key
 
-    def raw_results(
-        self,
-        query: str,
-        max_results: Optional[int] = 3,
-        search_depth: Optional[str] = "advanced",
-        include_domains: Optional[list[str]] = [],
-        exclude_domains: Optional[list[str]] = [],
-        include_answer: Optional[bool] = False,
-        include_raw_content: Optional[bool] = False,
-        include_images: Optional[bool] = False,
-    ) -> dict:
+    def raw_results(self, params: dict[str, Any]) -> dict:
         """
         Retrieves raw search results from the Tavily Search API.
 
         Args:
-            query (str): The search query.
-            max_results (int, optional): The maximum number of results to retrieve. Defaults to 3.
-            search_depth (str, optional): The search depth. Defaults to "advanced".
-            include_domains (List[str], optional): The domains to include in the search. Defaults to [].
-            exclude_domains (List[str], optional): The domains to exclude from the search. Defaults to [].
-            include_answer (bool, optional): Whether to include answer in the search results. Defaults to False.
-            include_raw_content (bool, optional): Whether to include raw content in the search results. Defaults to False.
-            include_images (bool, optional): Whether to include images in the search results. Defaults to False.
+            params (Dict[str, Any]): The search parameters.
 
         Returns:
             dict: The raw search results.
 
         """
-        params = {
-            "api_key": self.api_key,
-            "query": query,
-            "max_results": max_results,
-            "search_depth": search_depth,
-            "include_domains": include_domains,
-            "exclude_domains": exclude_domains,
-            "include_answer": include_answer,
-            "include_raw_content": include_raw_content,
-            "include_images": include_images,
-        }
+        params["api_key"] = self.api_key
         response = requests.post(f"{TAVILY_API_URL}/search", json=params)
```

Comment:
`includes_domains` and `exclude_domains` does not work here, it should be an array but got string instead, below could make sense I guess.
```python
params['exclude_domains'] = params.get('exclude_domains', '').split()
```